### PR TITLE
🧹 Fix layering issues with embedded languages interfaces

### PR DESCRIPTION
**Pull request by sharwell**
* Eliminate the distinction between `IEmbeddedLanguageEditorFeaturesProvider`, `IEmbeddedLanguageFeaturesProvider`, and `IEmbeddedLanguagesProvider` in the MEF catalog
* Eliminate the use of `Instance` members to bypass the declared `ServiceLayer` layering for features
* Eliminate the shadowing of `Language` properties with different contents depending on which service the property is accessed through

### CODE REVIEW

File: src/EditorFeatures/CSharp/EmbeddedLanguages/CSharpEmbeddedLanguageEditorFeaturesProvider.cs [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
 using Microsoft.CodeAnalysis.CSharp.Features.EmbeddedLanguages;
 using Microsoft.CodeAnalysis.Editor.EmbeddedLanguages;
-using Microsoft.CodeAnalysis.Features.EmbeddedLanguages;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
 using Microsoft.CodeAnalysis.Host.Mef;
 
 namespace Microsoft.CodeAnalysis.CSharp.Editor.EmbeddedLanguages
 {
-    [ExportLanguageService(typeof(IEmbeddedLanguageEditorFeaturesProvider), LanguageNames.CSharp), Shared]
+    [ExportLanguageService(typeof(IEmbeddedLanguagesProvider), LanguageNames.CSharp, ServiceLayer.Editor), Shared]
```
Comment (heejaechang) [inline src/EditorFeatures/CSharp/EmbeddedLanguages/CSharpEmbeddedLanguageEditorFeaturesProvider.cs]: IEmbededLanguagesProvider seems completely new pattern than anything else we have? I guess people were okay introducing this new concept?

Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharp/EmbeddedLanguages/CSharpEmbeddedLanguageEditorFeaturesProvider.cs]: @heejaechang Can you clarify?  This is a pretty normal language service, similar to how we've plugged in a lot of language specific behaviors.

Comment (sharwell) [reply src/EditorFeatures/CSharp/EmbeddedLanguages/CSharpEmbeddedLanguageEditorFeaturesProvider.cs]: > IEmbededLanguagesProvider seems completely new pattern than anything else we have? I guess people were okay introducing this new concept?

This was part of a different review. The issue addressed here is the separate export of three related interfaces, where the code should have uniformly exported the base interface.

**Pull request merged**